Userland/Libraries/LibGfx/ImageFormats/WebPLosslessShared.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024, Nico Weber <thakis@chromium.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibCompress/Deflate.h>
+
+namespace Gfx {
+
+// WebP-lossless's CanonicalCodes are almost identical to deflate's.
+// One difference is that codes with a single element in webp-lossless consume 0 bits to produce that single element,
+// while they consume 1 bit in Compress::CanonicalCode. This class wraps Compress::CanonicalCode to handle the case
+// where the codes contain just a single element, and dispatches to Compress::CanonicalCode else.
+class CanonicalCode {
+public:
+    CanonicalCode()
+        : m_code(0)
```

Comment:
I think that the recommendation is to default initialize member at their definition and not in the constructor. And you'll be able to `= default` the constructor as well.

Also, I'm quite confused by this commit, you say that it moves code, but I can only see additions.

Reply:
Excellent catch about the confusion. I had a merge conflict locally and didn't resolve it correctly. Now it should be better.

Re initialization: It's a variant; I think for variants you have to pick which variant you want to initialize.

Reply:
I know that we are speaking of initialization in C++, but how different that would be from initializing it in the constructor?

If it wasn't clear, that's what I meant (it compiles):

```diff
diff --git a/Userland/Libraries/LibGfx/ImageFormats/WebPLoaderLossless.cpp b/Userland/Libraries/LibGfx/ImageFormats/WebPLoaderLossless.cpp
index 89954d9171..1dcd6a8fd2 100644
--- a/Userland/Libraries/LibGfx/ImageFormats/WebPLoaderLossless.cpp
+++ b/Userland/Libraries/LibGfx/ImageFormats/WebPLoaderLossless.cpp
@@ -57,10 +57,7 @@ namespace {
 // where the codes contain just a single element, and dispatches to Compress::CanonicalCode else.
 class CanonicalCode {
 public:
-    CanonicalCode()
-        : m_code(0)
-    {
-    }
+    CanonicalCode() = default;
 
     static ErrorOr<CanonicalCode> from_bytes(ReadonlyBytes);
     ErrorOr<u32> read_symbol(LittleEndianInputBitStream&) const;
@@ -76,7 +73,7 @@ private:
     {
     }
 
-    Variant<u32, Compress::CanonicalCode> m_code;
+    Variant<u32, Compress::CanonicalCode> m_code { 0 };
 };
 
 ErrorOr<CanonicalCode> CanonicalCode::from_bytes(ReadonlyBytes bytes)
```

Reply:
That does look a lot nicer if you write it out like that. Thanks, done :)

(I made you the author of that commit.)